public_html/wp-content/plugins/wc-post-types/inc/rest-api.php

```diff
@@ -32,6 +34,49 @@ function register_sponsor_post_meta() {
 	);
 }
 
+/**
+ * Registers post meta to the Speaker post type.
+ *
+ * @return void
+ */
+function register_speaker_post_meta() {
+	register_post_meta(
+		'wcb_speaker',
+		'_wcpt_user_id',
+		array(
+			'type'         => 'integer',
+			// This is not set directly, but is set as a result of `_wcpt_user_name`.
+			'show_in_rest' => false,
+			'single'       => true,
+		)
+	);
+	register_post_meta(
+		'wcb_speaker',
+		'_wcpt_user_name',
+		array(
+			'type'              => 'string',
+			'single'            => true,
+			'show_in_rest'      => true,
+			'sanitize_callback' => function( $value ) {
+				$wporg_user = wcorg_get_user_by_canonical_names( $value );
+				if ( ! $wporg_user ) {
+					return '';
+				}
+				return $wporg_user->user_login;
+			},
+		)
+	);
+	register_post_meta(
+		'wcb_speaker',
+		'_wcb_speaker_email',
+		array(
+			'type'         => 'string',
+			'show_in_rest' => true,
```

Comment:
It looks like this exposes the speaker's email address in responses to unauthenticated requests to the `speakers` endpoint. Limiting the context to `edit` seems to avoid this:

```suggestion
			'show_in_rest' => array(
				'schema' => array(
					'context' => array( 'edit' ),
				),
			),
```